core/src/main/java/org/apache/calcite/sql/fun/SqlCaretNegationOperator.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.calcite.sql.fun;
+
+import org.apache.calcite.sql.SqlCall;
+import org.apache.calcite.sql.SqlKind;
+import org.apache.calcite.sql.SqlPrefixOperator;
+import org.apache.calcite.sql.SqlWriter;
+
+/**
+ * SqlCaretNegationOperator represents the "^" operator that occurs before a
+ * binary row expression.
+ */
+public class SqlCaretNegationOperator extends SqlPrefixOperator {
+
+  public SqlCaretNegationOperator() {
+    super("^", SqlKind.CARET_NEGATION, 26, null, null, null);
+  }
+
+  @Override public void unparse(SqlWriter writer, SqlCall call, int leftPrec,
+      int rightPrec) {
+    writer.keyword(getName());
+    writer.setNeedWhitespace(false);
+    call.operand(0).unparse(writer, getLeftPrec(), getRightPrec());
```

Comment:
What is the difference between calling getLeftPrec() and just passing in the leftPrec parameter which was passed to this method?

Reply:
I think I added those on accident, it works with leftPrec/rightPrec so I'll switch those calls to variables since that is what's usually used